src/sentry/integrations/msteams/utils.py

```diff
@@ -92,17 +94,20 @@ def get_channel_id(organization, integration_id, name):
     return None
 
 
-def send_incident_alert_notification(action, incident, metric_value, new_status: IncidentStatus):
+def send_incident_alert_notification(
+    action: AlertRuleTriggerAction,
+    incident: Incident,
+    metric_value: int | None,
+    new_status: IncidentStatus,
+) -> None:
     from .card_builder import build_incident_attachment
 
-    channel = action.target_identifier
-    integration = action.integration
     attachment = build_incident_attachment(incident, new_status, metric_value)
-    client = MsTeamsClient(integration)
-    try:
-        client.send_card(channel, attachment)
-    except ApiError:
-        logger.info("rule.fail.msteams_post", exc_info=True)
+    integration_service.send_msteams_incident_alert_notification(
```

Comment:
Maybe a few, but not as a general pattern as far as I can tell.

This one came up organically because of the special `sentry.integrations.msteams.utils.send_incident_alert_notification` function and its `MsTeamsClient` class (and only then because `MsTeamsClient` is tightly coupled to the `Integration` model).

Reply:
Fair. We can move forward as is. Integrations will still need a bunch of rework and if this method has ended up in the wrong place we can adjust later.